pkg/lib/indexer/indexer.go

```diff
@@ -283,3 +291,119 @@ func write(dockerfileText, outDockerfile string, logger *logrus.Entry) error {
 
 	return nil
 }
+
+// ExportFromIndexRequest defines the parameters to send to the ExportFromIndex API
+type ExportFromIndexRequest struct {
+	Index         string
+	Operator      string
+	DownloadPath  string
+	ContainerTool string
+}
+
+// ExportFromIndex is an aggregate API used to specific operators from
+// an index image
+func (i ImageIndexer) ExportFromIndex(request ExportFromIndexRequest) error {
+	databaseFile := defaultDatabaseFile
+
+	// set a temp directory
+	workingDir, err := ioutil.TempDir("./", "index_tmp")
+	if err != nil {
+		return err
+	}
+	defer os.RemoveAll(workingDir)
+
+	// Pull the fromIndex
+	if request.Index != "" {
+		i.Logger.Infof("Pulling previous image %s to get metadata", request.Index)
+
+		// Get the old index image's dbLocationLabel to find this path
+		labels, err := i.LabelReader.GetLabelsFromImage(request.Index)
+		if err != nil {
+			return err
+		}
+		if dbLocation, ok := labels[containertools.DbLocationLabel]; ok {
+			i.Logger.Infof("Previous db location %s", dbLocation)
+
+			// extract the database to the file
+			err = i.ImageReader.GetImageData(request.Index, workingDir)
+			if err != nil {
+				return err
+			}
+
+			databaseFile = path.Join(workingDir, dbLocation)
+		}
+	} else {
+		databaseFile = path.Join(workingDir, databaseFile)
+	}
+
+	db, err := sql.Open("sqlite3", databaseFile)
+	if err != nil {
+		return err
+	}
+	defer db.Close()
+
+	dbQuerier := sqlite.NewSQLLiteQuerierFromDb(db)
+	if err != nil {
+		return err
+	}
+
+	bundles, err := dbQuerier.GetImagesForPackage(context.TODO(), request.Operator)
+	if err != nil {
+		return err
+	}
+
+	var errs []error
+
+	for _, bundleImage := range bundles {
+		loader := sqlite.NewSQLExporterForImage(dbQuerier, bundleImage, request.DownloadPath, request.ContainerTool)
+		if err := loader.Export(); err != nil {
+			err = fmt.Errorf("error exporting bundle from image: %s", err)
+			errs = append(errs, err)
+		}
+	}
+
+	// generate package.yaml
+	defaultChannel, err := dbQuerier.GetDefaultChannelForPackage(context.TODO(), request.Operator)
+	if err != nil {
+		return err
+	}
+
+	channelList, err := dbQuerier.ListChannels(context.TODO(), request.Operator)
+	if err != nil {
+		return err
+	}
+
+	channels := []pregistry.PackageChannel{}
```

Comment:
typo: `pregistry`?

Reply:
No I think this is intentionally imported as `pregistry` to avoid collision with the overall `registry` package and points to `"github.com/operator-framework/operator-registry/pkg/registry"`